src/lib/tls/tls_session.cpp

```diff
@@ -428,12 +428,12 @@ Session Session::decrypt(std::span<const uint8_t> in, const SymmetricKey& key) {
          throw Decoding_Error("Encrypted session too short to be valid");
       }
 
-      // TODO: replace those raw pointers with sub-spans into `in`
-      const uint8_t* magic = in.data();
-      const uint8_t* key_name = magic + TLS_SESSION_CRYPT_MAGIC_LEN;
-      const uint8_t* key_seed = key_name + TLS_SESSION_CRYPT_KEY_NAME_LEN;
-      const uint8_t* aead_nonce = key_seed + TLS_SESSION_CRYPT_AEAD_KEY_SEED_LEN;
-      const uint8_t* ctext = aead_nonce + TLS_SESSION_CRYPT_AEAD_NONCE_LEN;
+      BufferSlicer sub(in);
+      const auto magic = sub.take(TLS_SESSION_CRYPT_MAGIC_LEN).data();
+      const auto key_name = sub.take(TLS_SESSION_CRYPT_KEY_NAME_LEN).data();
+      const auto key_seed = sub.take(TLS_SESSION_CRYPT_AEAD_KEY_SEED_LEN).data();
+      const auto aead_nonce = sub.take(TLS_SESSION_CRYPT_AEAD_NONCE_LEN).data();
+      const auto ctext = sub.take(TLS_SESSION_CRYPT_HDR_LEN).data();
       const size_t ctext_len = in.size() - TLS_SESSION_CRYPT_HDR_LEN;  // includes the tag
```

Comment:
```suggestion
      auto ctext = sub.copy_as_secure_vector(sub.remaining());
```

... and then just use the `ctext` variable in `aead->finish()` down below straight away without copying into `buf` before.